src/OMSimulatorLib/Scope.cpp

```diff
@@ -1969,6 +1973,130 @@ oms_status_enu_t oms2::Scope::setMasterAlgorithm(const ComRef& cref, const std::
   return oms_status_ok;
 }
 
+oms_status_enu_t oms2::Scope::addEventIndicator(const oms2::SignalRef& signal)
+{
+  logTrace();
+  ComRef cref=signal.getCref();
+
+  if (!cref.isIdent()) {
+    ComRef modelCref=cref.first();
+    Model* model=getModel(modelCref);
+    if (!model)
+    {
+      logError("[oms2::Scope::addEventIndicator] failed");
+      return oms_status_error;
+    }
+    model -> addEventIndicator(signal);
+    
+    return oms_status_ok;
+  }
+
+  return oms_status_error;
+}
+
+oms_status_enu_t oms2::Scope::addTimeIndicator(const oms2::SignalRef& signal)
+{
+  logTrace();
+  ComRef cref=signal.getCref();
+
+  if (!cref.isIdent()) {
+    ComRef modelCref=cref.first();
+    Model* model=getModel(modelCref);
+    if (!model)
+    {
+      logError("[oms2::Scope::addTimeIndicator] failed");
+      return oms_status_error;
+    }
+    model -> addTimeIndicator(signal);
+    
+    return oms_status_ok;
+  }
+
+  return oms_status_error;
+}
+
+oms_status_enu_t oms2::Scope::addStaticValueIndicator(const oms2::SignalRef& signal, double lower, double upper, double stepSize) 
+{
+  logTrace();
+  ComRef cref=signal.getCref();
+
+  if (!cref.isIdent()) {
+    ComRef modelCref=cref.first();
+    Model* model=getModel(modelCref);
+    if (!model)
+    {
+      logError("[oms2::Scope::addStaticValueIndicator] failed");
+      return oms_status_error;
+    }
+    model -> addStaticValueIndicator(signal,lower,upper,stepSize);
+    
+    return oms_status_ok;
+  }
+
+  return oms_status_error;
+}
+
+oms_status_enu_t oms2::Scope::addDynamicValueIndicator(const oms2::SignalRef& signal, const oms2::SignalRef& lower, const oms2::SignalRef& upper, double stepSize)
+{
+  logTrace();
+  ComRef cref=signal.getCref();
+
+  if (!cref.isIdent()) {
+    ComRef modelCref=cref.first();
+    Model* model=getModel(modelCref);
+    if (!model)
+    {
+      logError("[oms2::Scope::addDynamicValueIndicator] failed");
+      return oms_status_error;
+    }
+    model -> addDynamicValueIndicator(signal,lower,upper,stepSize);
+    
+    return oms_status_ok;
+  }
+
+  return oms_status_error;
+}
+   
+
+oms_status_enu_t oms2::Scope::setMinimalStepSize(const ComRef& cref, double min)
+{
+  logTrace();
+
+  if (cref.isIdent()) {
+    Model* model = getModel(cref);
+    if (!model)
+    {
+      logError("[oms2::Scope::setMinimalStepSize] failed");
+      return oms_status_error;
+    }
```

Comment:
```C++
if (!model)
  return logError_ModelNotInScope(cref.toString());
```

The name of the method itself is now automatically added to `logError` messages . And `logError` returns `oms_status_error` which improves the readability of the code.

There are actually more error messages that can be improved.